src/zarr/core/array.py

```diff
@@ -611,7 +609,7 @@ async def _create_v2(
         if not exists_ok:
             await ensure_no_existing_node(store_path, zarr_format=2)
         if order is None:
-            order = "C"
+            order = config.get("array.order", "C")
```

Comment:
Suppose we change the key `array.order` to `array_indexing.order` in the config, this will happily return "C", when the correct behavior should be an error. Therefore I think we should define the default value higher up than this function, and this function should have a hard dependency on the schema of the config.

Reply:
I'm not sure I follow. What's the ask here?

Reply:
How much higher? We could require passing  `order` into `Array.create` rather than allowing this to be a nullable field. Or we could do the `config` check there.

Reply:
would it be possible to put the default in the config? I'm guessing we will have a lot of defaults (e.g., codecs), so would probably make sense to put them all in the same place

Reply:
looks like the [defaults are already in the config](https://github.com/zarr-developers/zarr-python/blob/37fde7defadf13b6894487987e99e098f027dcf6/src/zarr/core/config.py#L45), in which case I think we can just remove any "default to C" behavior in `_create_v2`

Reply:
okay. I think the final state should be what you are after. Because the v2 array metadata requires `order`, we do have to check the config setting if no `order` was passed, but we no longer apply a default on top of that.